tests/unit/lxd/test_launcher.py

```diff
@@ -832,3 +922,73 @@ def test_set_id_map_all_options(fake_base_instance, mock_lxc, mocker):
             remote="test-remote",
         )
     ]
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="unsupported on windows")
+@pytest.mark.parametrize(
+    "map_user_uid, actual_uid, expected_uid, expected_result",
+    [
+        # return True if an id map is not expected and there is no id map
+        (False, None, None, True),
+        # return False if an id map is expected and there is no id map
+        (True, None, 5678, False),
+        # return False if id map is not expected and there is an id map
+        (False, 1234, None, False),
+        # return True if an id map is expected and the uid matches
+        (True, 1234, 1234, True),
+        # return False if an id map is expected and the uid does not match
+        (True, 1234, 5678, False),
+        # return True if an id map is expected and the uid matches the current users uid
+        (True, 101, None, True),
+    ],
```

Comment:
This is fantastic, clear parametrization right here.

Reply:
Thanks!